File: exo3/exo3.php
<?php
#comptage du nombre de caractere 
#author sebastien
#beWeb
function rechercheCarac($word,$carac){

	$argv[1] = $word;
	$argv[2] = $carac;
	$resultat = 0;
	#initialisation de l'index donc index 0
	$i = 0;
	
	#tant que index inferieur a la longueur du mot cela exectute le code ci-dessous
	while($i < strlen($word)){
		#si a l'index ou je me trouve c'est bien un "n" j'incremente de 1
		if($word[$i] == $carac){
			$resultat++;
		}
	#et je passe a l'index suivant
	$i++;
	}
	echo "Le nombre de caractereRecherche est : " .$resultat."\r"."\n";
}
#on recupere les parametre passer dans le terminal
rechercheCarac($argv[1],$argv[2])
?>

File: exo3/exo3.rb
#comptage du nombre de caractere 
#author sebastien
#beWeb

def countChar(word,char)
	word = ARGV[0]
	char = ARGV[1]
	resultat = 0
	i = 0

	#tant que index inferieur a la longueur du mot cela exectute le code ci-dessous
	while i < word.length
		#si a l'index ou je me trouve c'est bien un "n" j'incremente de 1
		if word[i] == char
			resultat += 1
		end
	#et je passe a l'index suivant
	i += 1
	end
	print "Le nombre de #{char} dans #{word} est: #{resultat}\r\n"

end

#on recupere les parametre passer dans le terminal
countChar(ARGV[0],ARGV[1])
